core/src/main/java/tech/pegasys/web3signer/core/service/http/handlers/keymanager/imports/ImportKeystoresHandler.java

```diff
@@ -221,43 +223,25 @@ private void handleInvalidRequest(final RoutingContext routingContext, final Exc
     routingContext.fail(BAD_REQUEST, e);
   }
 
-  public void createKeyStoreYamlFileAt(
-      final String fileName, final String jsonKeystoreData, final String password)
-      throws IOException {
-
-    final Path yamlFile = keystorePath.resolve(fileName + ".yaml");
-
-    final String keystoreFileName = fileName + ".json";
-    final Path keystoreFile = yamlFile.getParent().resolve(keystoreFileName);
-    createTextFile(keystoreFile, jsonKeystoreData);
-
-    final String passwordFilename = fileName + ".password";
-    final Path passwordFile = yamlFile.getParent().resolve(passwordFilename);
-    createTextFile(passwordFile, password);
-
-    final FileKeyStoreMetadata data =
-        new FileKeyStoreMetadata(keystoreFile, passwordFile, KeyType.BLS);
-    createYamlFile(yamlFile, data);
-  }
-
-  private void createTextFile(final Path keystoreFile, final String jsonKeystoreData)
-      throws IOException {
-    Files.writeString(keystoreFile, jsonKeystoreData, StandardCharsets.UTF_8);
-  }
-
-  private void createYamlFile(final Path filePath, final FileKeyStoreMetadata signingMetadata)
-      throws IOException {
-    YAML_OBJECT_MAPPER.writeValue(filePath.toFile(), signingMetadata);
-  }
-
   private void removeSignersAndCleanupImportedKeystoreFiles(final List<String> pubkeys) {
     for (String pubkey : pubkeys) {
       try {
         artifactSignerProvider.removeSigner(pubkey).get();
-        Files.deleteIfExists(keystorePath.resolve(pubkey + ".yaml"));
-      } catch (Exception e) {
-        LOG.error("Failed to cleanup imported keystore file for key: " + pubkey);
+      } catch (final InterruptedException | ExecutionException e) {
+        LOG.warn("Unable to remove signer for {} due to {}", pubkey, e.getMessage());
       }
+
+      deleteFile(keystorePath.resolve(pubkey + METADATA_YAML_EXTENSION));
+      deleteFile(keystorePath.resolve(pubkey + KEYSTORE_JSON_EXTENSION));
+      deleteFile(keystorePath.resolve(pubkey + KEYSTORE_PASSWORD_EXTENSION));
```

Comment:
Don't think deleting the files is necessary anymore now that the KeyStoreFileManager cleans up on failure

Reply:
This might still be required if error happens outside create operation ... since its 'deleteIfExist' it won't cause unnecessary exceptions.